src/Dictionaries/getDictionaryConfigurationFromAST.cpp

```diff
@@ -609,6 +609,12 @@ getDictionaryConfigurationFromAST(const ASTCreateQuery & query, ContextPtr conte
 
     bool complex = DictionaryFactory::instance().isComplex(dictionary_layout->layout_type);
 
+    if (pk_attrs.size() > 1 && !complex
```

Comment:
Can you please add functional tests for this feature ?

Reply:
ok, I will try